tools/releases/dockerfiles/Dockerfile.kuma-dp

```diff
@@ -1,28 +1,20 @@
-ARG BASE_IMAGE_ARCH=amd64
-
 FROM debian:11 as envoy
 ARG ENVOY_VERSION
 ARG ARCH
 
-ADD /build/artifacts-linux-$ARCH/envoy/envoy-$ENVOY_VERSION-alpine /envoy
+COPY /build/artifacts-linux-$ARCH/envoy/envoy-$ENVOY_VERSION-alpine /envoy
 
 RUN apt-get update && \
-    apt-get install -y libcap2-bin
-RUN setcap cap_net_bind_service+ep /envoy
+    apt-get install --no-install-recommends -y libcap2-bin=1:2.44-1 && \
+    setcap cap_net_bind_service+ep /envoy
```

Comment:
This modifies the binary?

Reply:
no it does not, so this never worked

```
root@9833f232441e:/# md5sum /bin/ls
fffefca59f1dcb04e318b6b26fa1b50e  /bin/ls
root@9833f232441e:/# setcap cap_net_bind_service+ep /bin/ls
root@9833f232441e:/# md5sum /bin/ls
fffefca59f1dcb04e318b6b26fa1b50e  /bin/ls
```

Reply:
"All the capabilities for processes and threads are stored in the status file under the process/thread directory in the /proc file system."

@michaelbeaumont - this was introduced in your [PR](https://github.com/kumahq/kuma/pull/4535/files#diff-d45a2aaea699d707bf64399c4dd7ee5d019b7a78a92ae8cec568f40a96bd737eR11) - did you check that it was needed?

Reply:
Yes, it's needed and it does work.

> All the capabilities for processes and threads are stored in the status file under the process/thread directory in the /proc file system.

I think this is from [this article](https://tbhaxor.com/understanding-linux-capabilities/#where-are-capabilities-records-stored) but I believe it refers to the capabilities in the context of a running process.

From the same article:

> Where are Capabilities Records Stored?
Extended permissions such as access control lists set by [setfacl](https://man7.org/linux/man-pages/man1/setfacl.1.html) and capability flags set by [setcap](https://man7.org/linux/man-pages/man8/setcap.8.html) which use [setxattr(2)](https://man7.org/linux/man-pages/man2/setxattr.2.html) are stored in the same place as traditional permissions and set[ug]id flags set by [chmod](https://man7.org/linux/man-pages/man1/chmod.1.html) –  in the file's inode.

or [`man capabilities`](https://man7.org/linux/man-pages/man7/capabilities.7.html):

> File capabilities
       Since kernel 2.6.24, the kernel supports associating capability
       sets with an executable file using [setcap(8)](https://man7.org/linux/man-pages/man8/setcap.8.html).  The file
       capability sets are stored in an extended attribute (see
       [setxattr(2)](https://man7.org/linux/man-pages/man2/setxattr.2.html) and [xattr(7)](https://man7.org/linux/man-pages/man7/xattr.7.html)) named security.capability.  Writing to
       this extended attribute requires the CAP_SETFCAP capability.  The
       file capability sets, in conjunction with the capability sets of
       the thread, determine the capabilities of a thread after an
       [execve(2)](https://man7.org/linux/man-pages/man2/execve.2.html).

`md5sum` won't include these but they are copied by docker [(at least when using buildkit, docker docs are terribly incomplete)](https://wildwolf.name/multi-stage-docker-builds-and-xattrs/) along with the executable.

Reply:
Cool worth adding a comment with this info for the next folks that look at this :)